eth2/types/src/slot_height.rs

```diff
@@ -0,0 +1,311 @@
+// Copyright 2019 Sigma Prime Pty Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+use crate::slot_epoch::{Epoch, Slot};
+use serde_derive::Serialize;
+use slog;
+use ssz::{hash, ssz_encode, Decodable, DecodeError, Encodable, SszStream, TreeHash};
+use std::cmp::{Ord, Ordering};
+use std::fmt;
+use std::hash::{Hash, Hasher};
+use std::ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Rem, Sub, SubAssign};
+
+macro_rules! impl_from_into_u64 {
```

Comment:
Would be nice if we can avoid duplicating all of these macros. Are they functionally different?

Reply:
Yeah, went to import them from the other one. Using export macro's.. but seems it wanted all the macro's to be in their own file. i.e seperate to `slot_epoch`. 
Happy for suggestions on this.

Reply:
You can definitely share macros inside a module but I don't know the intricacies of it, especially since Rust 2018.

I can have a go at it if you like.

https://stackoverflow.com/a/31749071

Reply:
Oh sorry, just re-read your question.

Maybe `slot_epoch_macros.rs`? Not the nicest name but better than reproducing the code, IMO

Reply:
Oh noticed you already commented on this above.